### PR TITLE
fix(ingest/bigquery): Fix shard regexp to match without underscore as well

### PR DESCRIPTION
**Pull request by treff7es**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/tests/unit/test_bigquery_source.py [hunk 1]
```diff
@@ -760,11 +760,14 @@ def test_gen_view_dataset_workunits(
         ("project.dataset.table_20231215", "project.dataset.table", "20231215"),
         ("project.dataset.table_2023", "project.dataset.table_2023", None),
         # incorrectly handled special case where dataset itself is a sharded table if full name is specified
-        ("project.dataset.20231215", "project.dataset.20231215", None),
+        ("project.dataset.20231215", "project.dataset.20231215", "20231215"),
+        ("project1.dataset2.20231215", "project1.dataset2.20231215", "20231215"),
         # Cases with Just the table name as input
         ("table", "table", None),
-        ("table20231215", "table20231215", None),
+        ("table20231215", "table", "20231215"),
```
Comment (hsheth2) [inline metadata-ingestion/tests/unit/test_bigquery_source.py:767]: wow all these tests for a single regex

**Review by hsheth2 — COMMENTED** [review]
assuming all the regexes are correct, this looks good to me

**Review by hsheth2 — APPROVED** [review]
assuming all the regexes are correct, this looks good to me

File: metadata-ingestion/tests/unit/test_bigquery_source.py (hunk 1 above)
Comment (treff7es) [reply metadata-ingestion/tests/unit/test_bigquery_source.py:767]: Haha, yes, it is super crazy but very well-detailed

**Comment by hsheth2** [comment]
@treff7es there's a test error

```
FAILED tests/unit/test_bigqueryv2_usage_source.py::test_bigquery_table_sanitasitation - AssertionError: assert False
 +  where False = <bound method BigqueryTableIdentifier.is_sharded_table of BigqueryTableIdentifier(project_id='project-1234', dataset='dataset-4567', table='foo_20222110', _BQ_SHARDED_TABLE_SUFFIX='_yyyymmdd')>()
 +    where <bound method BigqueryTableIdentifier.is_sharded_table of BigqueryTableIdentifier(project_id='project-1234', dataset='dataset-4567', table='foo_20222110', _BQ_SHARDED_TABLE_SUFFIX='_yyyymmdd')> = BigqueryTableIdentifier(project_id='project-1234', dataset='dataset-4567', table='foo_20222110', _BQ_SHARDED_TABLE_SUFFIX='_yyyymmdd').is_sharded_table
```

**Comment by treff7es** [comment]
> test_bigqueryv2_usage_source

thanks, the test was not correct (earlier it was an invalid data and we properly check it)

**Review by hsheth2 — APPROVED** [review]

**Pull request merged**